target/linux/ramips/dts/mt7621_zyxel_keenetic-giga-iii.dts

```diff
@@ -0,0 +1,301 @@
+			reg = <0x100000 0x80000>; // 512 KB
+			read-only;
+		};
+
+		/* U-Boot expects to find kernel image at 0x180000 */
+		partition@180000 {
+			label = "firmware";         // Firmware_1
+			reg = <0x180000 0x3E40000>; // 122 MB
+			/* For revert to stock: mtd -r -f write /tmp/fw_factory.bin firmware */
+
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "kernel";
+				reg = <0x0 0x400000>; // 4 MB
+			};
+
+			ubiconcat0: partition@400000 {
+				label = "ubiconcat0";
+				reg = <0x400000 0x3A40000>; // 58 MB
+			};
+		};
+
+		partition@3fc0000 { // UBoot algo: (0x8000000 - 0x80000) / 2 = 0x3FC0000
+			label = "u-state";
+			reg = <0x3fc0000 0x80000>; // 512 KB
+			read-only;
+		};
+
+		partition@4040000 {
+			label = "u-config-res";
+			reg = <0x4040000 0x80000>; // 512 KB
+			read-only;
+		};
+
+		partition@40c0000 {
+			label = "factory-res";     // RF-EEPROM_res
+			reg = <0x40c0000 0x80000>; // 512 KB
+			read-only;
+		};
+
+		ubiconcat1: partition@4140000 {
+			label = "ubiconcat1";        // Firmware_2
+			reg = <0x4140000 0x3AC0000>; // 58 MB
+		};
+
+		/* U-Boot used latest 32 BLOCK for factory bbt table */
+		partition@7C00000 { // 128MiB - 32*128KiB = 0x7C00000
+			label = "mtk-nand-bbt";
+			reg = <0x7C00000 0x400000>; // 4 MB
+			read-only;
+		};
+	};
+};
+
+&xhci {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 { // MT7612EN wifi5g
+	wifi0: mt76@0,0 {  
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		nvmem-cells = <&macaddr_factory_8004>;
+		nvmem-cell-names = "mac-address";
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&pcie1 { // MT7602EN wifi2g
+	wifi1: mt76@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		nvmem-cells = <&macaddr_factory_4>;
+		nvmem-cell-names = "mac-address";
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_28>;
+	nvmem-cell-names = "mac-address";
+};
+
+&switch0 {
+	ports {
+		port@0 {
+			status = "okay";
+			label = "lan4";
+		};
+
+		port@1 {
+			status = "okay";
+			label = "lan3";
+		};
+
+		port@2 {
+			status = "okay";
+			label = "lan2";
+		};
+
+		port@3 {
+			status = "okay";
+			label = "lan1";
+		};
+
+		port@4 {
+			status = "okay";
+			label = "wan";
+			nvmem-cells = <&macaddr_factory_28>;
+			nvmem-cell-names = "mac-address";
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "uart3", "jtag", "wdt";
+		function = "gpio";
+	};
+};
+
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	macaddr_factory_4: macaddr@4 { /* LAN and wifi2 mac (wan - 1) */
+		reg = <0x4 0x6>;
+	};
+
+	macaddr_factory_28: macaddr@28 { /* WAN mac (LABEL) */
+		reg = <0x28 0x6>;
+	};
+
+	macaddr_factory_8004: macaddr@8004 { /* wifi5 mac (wan + 1) */
+		reg = <0x8004 0x6>;
+	};
```

Comment:
This last block can be dropped, since we get it from mtd-eeprom.

Reply:
Although not currently used, it should be kept for information.